packages/database/lib/schema.ts

```diff
@@ -1,5 +1,71 @@
-import { pgTable, text } from 'drizzle-orm/pg-core';
+import { bigint, index, pgTable, primaryKey, smallint, smallserial, varchar } from 'drizzle-orm/pg-core';
+
+// TODO: table(s) representing order book for each market.
+
+// https://docs.rs/solana-program/latest/src/solana_program/pubkey.rs.html#24
+const MAX_PUBKEY_B58_STR_LEN = 44;
+const pubkey = (columnName: string) => varchar(columnName, {length: MAX_PUBKEY_B58_STR_LEN});
+
+const MAX_TRANSACTION_B58_STR_LEN = 88;
+const transaction = (columnName: string) => varchar(columnName, {length: MAX_TRANSACTION_B58_STR_LEN});
+
+const tokenAmount = (columnName: string) => bigint(columnName, {mode: 'bigint'});
+
+// time represented in milliseconds since epoch
+const timestamp = (columnName: string) => bigint(columnName, {mode: 'bigint'});
 
 export const proposals = pgTable('proposals', {
-    id: text('id').primaryKey(),
+  proposalId: pubkey('proposal_id').primaryKey(),
+  autocratVersion: smallint('autocrat_version').notNull()
 });
+
+export enum MarketType {
+  OPEN_BOOK = 'OPEN_BOOK'
+}
+
+type NonEmptyList<E> = [E, ...E[]];
+
+export const markets = pgTable('markets', {
+  marketId: pubkey('market_id').primaryKey(),
+  // may be null as market might not be tied to any one proposal (ex: the META spot market)
+  proposalId: pubkey('proposal_id').references(() => proposals.proposalId),
+  marketType: varchar('market_type', {enum: Object.values(MarketType) as NonEmptyList<MarketType>}).notNull(),
+  baseCurrencyMint: pubkey('base_currency_mint').references(() => tokens.tokenMint).notNull(),
+  quoteCurrencyMint: pubkey('quote_currency_mint').references(() => tokens.tokenMint).notNull(),
+});
+
+export const tokens = pgTable('tokens', {
+  tokenMint: pubkey('token_mint').primaryKey(),
+  name: varchar('name', {length: 30}).notNull(),
+  symbol: varchar('symbol', {length: 10}).notNull(),
+  decimals: smallserial('decimals').notNull()
+});
+
+export const trades = pgTable('trades', {
+  transactionId: transaction('transaction_id').primaryKey(),
+  slot: bigint('slot', {mode: 'bigint'}).notNull(),
+  block: bigint('block', {mode: 'bigint'}).notNull(),
+  time: timestamp('time').notNull(),
+  marketId: pubkey('market_id').references(() => markets.marketId).notNull(),
+  takerId: pubkey('taker_id').notNull(),
```

Comment:
I think indexing this and adding a makerId too....

Reply:
I thought on this a lot. I'll have the field be nullable for the case where it's an AMM. For order books it makes sense we can find the maker id.